tools/reaction_analyzer/README.md

```diff
@@ -0,0 +1,202 @@
+different ways: `async_header_sync_publish`, `sync_header_sync_publish` or `async_publish`. (`T` is the period of the
+lidar's output)
+
+![PointcloudPublisherType.png](media%2FPointcloudPublisherType.png)
+
+- `async_header_sync_publish`: It publishes the point cloud messages synchronously with asynchronous header times. It
+  means that each of the lidar's output will be published at the same time, but the headers of the point cloud messages
+  includes different timestamps because of the phase difference.
+- `sync_header_sync_publish`: It publishes the point cloud messages synchronously with synchronous header times. It
+  means that each of the lidar's output will be published at the same time, and the headers of the point cloud messages
+  includes the same timestamps.
+- `async_publish`: It publishes the point cloud messages asynchronously. It means that each of the lidar's output will
+  be published at different times.
+
+## Usage
+
+The common parameters you need to define for both running modes are `output_file_path`, `test_iteration`,
+and `reaction_chain` list. `output_file_path` is the output file path is the path where the results and statistics
+will be stored. `test_iteration` defines how many tests will be performed. The `reaction_chain` list is the list of the
+pre-defined topics you want to measure their reaction times.
+
+### Prepared Test Environment
+
+- Download the demonstration test map from the
+  link [here](https://github.com/tier4/AWSIM/releases/download/v1.1.0/nishishinjuku_autoware_map.zip). After
+  downloading,
+  extract the zip file and use its path as `[MAP_PATH]` in the following commands.
+
+#### Planning Control Mode
+
+- You need to define only Planning and Control nodes in the `reaction_chain` list. With the default parameters,
+  you can start to test with the following command:
+
+```bash
+ros2 launch reaction_analyzer reaction_analyzer.launch.xml running_mode:=planning_control vehicle_model:=sample_vehicle sensor_model:=sample_sensor_kit map_path:=[MAP_PATH]
+```
+
+After the command, the `simple_planning_simulator` and the `reaction_analyzer` will be launched. It will automatically
+start to test. After the test is completed, the results will be stored in the `output_file_path` you defined.
+
+#### Perception Planning Mode
+
+- Download the rosbag files from the Google Drive
+  link [here](https://drive.google.com/file/d/1_P-3oy_M6eJ7fk8h5CP8V6s6da6pPrBu/view?usp=sharing).
+- Extract the zip file and set the path of the `.db3` files to parameters `path_bag_without_object`
+  and `path_bag_with_object`.
+- Because custom sensor setup, you need to check out the following branches before launch the
+  reaction analyzer: For the `autoware_individual_params` repository, check out the
+  branch [here](https://github.com/brkay54/autoware_individual_params/tree/bk/reaction-analyzer-config).
```

Comment:
Yes you are right @mitsudome-r-san. Actually, main reason of the these branches is we need multiple lidar setup to test sensing pipeline accurately (to run concat and time sync process), however, current awsim sensor configuration is not supporting the multiple sensor currently. It might be planned to add multiple lidar in AWSIM sensor_kit. If it will be added, we can record a new rosbag to run the `reaction_analyzer` and we won't need to switch the branches in `awsim_sensor_kit_launch` and `autoware_individual_params`. I guess there would be some plans to add multiple lidar sensor setup to AWSIM. 

@xmfcx If there is a plan to add multiple lidar to AWSIM in near future, we can use branches in my remote (for awsim_sensor_kit_launch and autoware_individual_params) temporarily. After AWSIM sensor_kit updated, we can update the reaction_analyzer to be able to run with updated awsim_sensor_kit by default. 

What do you thing about this plan?

Reply:
@brkay54 When you were testing in your local environment, were you testing with customized AWSIM with multiple lidars? I think the default AWSIM binary doesn't have multiple LIDARs attached.

Reply:
> @brkay54 When you were testing in your local environment, were you testing with customized AWSIM with multiple lidars?

@mitsudome-r Yes, I was using customized AWSIM setup.

>  I think the default AWSIM binary doesn't have multiple LIDARs attached.

Yes, currently the setup is with 1 Lidar. Just I was assuming multiple lidar setup would be added to AWSIM in near future (maybe @xmfcx inform us).
My suggestion was lets merge this package with this AWSIM customization, after the AWSIM Setup is updated with multiple lidar setup, I can update the usage part of the readme with respect to latest update in AWSIM.

Reply:
remaining action items:
- create a pull request against https://github.com/RobotecAI/awsim_sensor_kit_launch/pulls and https://github.com/autowarefoundation/autoware_individual_params/pulls for multi-lidar support.
- Test it with AWF AWSIM main branch

Reply:
I run the some updated version of AWSIM with 3 - Lidar setup,  I couldn't see any issue. 

- https://github.com/autowarefoundation/AWSIM/pull/54#pullrequestreview-2008680579

After the PR merged, I am going to record the rosbags inside AWSIM to run the reaction_analyzer.

Reply:
I didn't realize the non-ground points on the road, it was my fault. I asked my colleague to help us to fix the transformations for sensors. 

https://github.com/autowarefoundation/autoware.universe/assets/45468306/791a0c65-519c-4468-aa2b-5ebfab171239

Reply:
@mozhoku Please help Berkay to check if the sensor calibration data in https://github.com/autowarefoundation/awsim_labs_sensor_kit_launch/tree/main/awsim_labs_sensor_kit_description/config match with the values in Unity.

Reply:
@brkay54 I've updated the right VLP16 params here:
- https://github.com/autowarefoundation/autoware_individual_params/pull/48

Reply:
cc @xmfcx @mitsudome-r, I updated the package w.r.t. new awsim_labs configuration. I tested both planning_control and perception_planning modes, I couldn't see any problem.

Current Scene Change for perception_planning mode:

![cap-2024-05-02-12-32-35-ezgif com-video-to-gif-converter](https://github.com/autowarefoundation/autoware.universe/assets/45468306/0603a24a-f0d8-4787-a0cf-ec363226c552)

It is ready for review again. If you have some time, could you check it?

Reply:
I will run them on my machine as well. Once confirmed, I think we can merge this PR.